networks/p2p/discover/discover_storage_simple.go

```diff
@@ -222,8 +221,8 @@ func (s *simpleStorage) closest(target common.Hash, nresults int) *nodesByDistan
 	// in the SimpleStorage. Change it
 	cNodes := &nodesByDistance{target: target}
 	nodes := s.shuffle(s.nodes)
-	if len(nodes) > s.max {
-		cNodes.entries = nodes[:s.max]
+	if len(nodes) > nresults {
+		cNodes.entries = nodes[:nresults]
```

Comment:
We might need to preserve the `s.max` limitation because it's uncertain if `s.nodes` growth is strictly capped by `s.max`.